src/renderer/components/playlist-info/playlist-info.js

```diff
@@ -79,49 +168,160 @@ export default defineComponent({
-          break
-        case 'openInvidious':
-          openExternalLink(invidiousUrl)
-          break
+    toggleCopyVideosPrompt: function (force = false) {
+      if (this.moreVideoDataAvailable && !force) {
+        showToast(this.$t('User Playlists.SinglePlaylistView.Toast["Some videos in the playlist are not loaded yet. Click here to copy anyway."]'), 5000, () => {
+          this.toggleCopyVideosPrompt(true)
+        })
+        return
       }
-    }
-  }
+
+      this.showAddToPlaylistPromptForManyVideos({
+        videos: this.videos,
+        newPlaylistDefaultProperties: { title: this.title },
+      })
+    },
+
+    savePlaylistInfo: function () {
+      if (this.newTitle === '') {
+        showToast(this.$t('User Playlists.SinglePlaylistView.Toast["Playlist name cannot be empty. Please input a name."]'))
+        return
+      }
+
+      const playlist = {
+        playlistName: this.newTitle,
+        protected: this.selectedUserPlaylist.protected,
+        description: this.newDescription,
+        videos: this.selectedUserPlaylist.videos,
+        _id: this.id,
+      }
+      try {
+        this.updatePlaylist(playlist)
+        showToast(this.$t('User Playlists.SinglePlaylistView.Toast["Playlist has been updated."]'))
+      } catch (e) {
+        showToast(this.$t('User Playlists.SinglePlaylistView.Toast["There was an issue with updating this playlist."]'))
+        console.error(e)
+      } finally {
+        this.exitEditMode()
+      }
+    },
+
+    enterEditMode: function () {
+      this.newTitle = this.title
+      this.newDescription = this.description
+      this.editMode = true
+
+      this.$emit('enter-edit-mode')
+
+      nextTick(() => {
+        // Some elements only present after rendering update
+        this.$refs.playlistTitleInput.focus()
+      })
+    },
+
+    exitEditMode: function () {
+      this.editMode = false
+
+      this.$emit('exit-edit-mode')
+    },
+
+    handleRemoveVideosOnWatchPromptAnswer: function (option) {
+      if (option === 'yes') {
+        const videosToWatch = this.selectedUserPlaylist.videos.filter((video) => {
+          return this.historyCacheById[video.videoId] == null
+        })
+
+        const videosRemoved = this.selectedUserPlaylist.videos.length - videosToWatch.length
+
+        if (videosRemoved === 0) {
+          showToast(this.$t('User Playlists.SinglePlaylistView.Toast["There were no videos to remove."]'))
+          this.showRemoveVideosOnWatchPrompt = false
+          return
+        }
+
+        const playlist = {
+          playlistName: this.title,
+          protected: this.selectedUserPlaylist.protected,
+          description: this.description,
+          videos: videosToWatch,
+          _id: this.id
+        }
+        try {
+          this.updatePlaylist(playlist)
+          showToast(this.$t('User Playlists.SinglePlaylistView.Toast["{videoCount} video(s) have been removed."]', {
```

Comment:
**suggestion (non-blocking):** use `tc` and update label with pluralization in lieu of "(s)".

Reply:
https://github.com/FreeTubeApp/FreeTube/pull/4234/commits/e7729eecc4c88a28c0385b267ac233603eeba006